trieste/acquisition/function.py

```diff
@@ -405,3 +408,93 @@ def probability_of_feasibility(
+
+    def predict_f_samples_with_reparametrisation_trick(self, model: ModelInterface, at: QueryPoints) -> tf.Tensor:
+        """
+        Returns samples according to the reparametrization trick. The sample size S is determined by eps,
+        the batch size B must be compatible between eps and at, N is the number of batches of query points
+        and L is the number of latent processes.
+        :param model:
+        :param at: batch of query points [N, D] or [N, B, D]
+        :return: a [S, N, B, L] tensor
+        """
+        # mean, cov = model.predict_f(at, full_cov=True, full_output_cov=True)
+
+        if len(at.shape) == 2:
+            mean, cov = model.predict(at)
+            # mean: [N, L]
+            # cov: [N, L]
+            samples = mean[None, :, None, :] + \
+                      tf.math.sqrt(cov)[None, :, None, :] * self.eps[:, None, ...]  # [S, N, B, L]
+        else:
+            assert at.shape[1] == self.eps.shape[1]
+
+            mean, cov = model.predict(at, full_cov=True)
+            # mean: [..., N, L]
+            # cov: [..., L, N, N]
+            mean_for_sample = tf.linalg.adjoint(mean)  # [..., L, N]
+            mean_shape = tf.shape(mean)
+            num_latent = mean_shape[-1]
+
+            jittermat = (
+                    tf.eye(num_latent, batch_shape=mean_shape[:-1], dtype=default_float()) * default_jitter()
+            )  # [..., N, D, D]
+
+            chol = tf.linalg.cholesky(cov + jittermat)  # [..., N, L, L]
+            samples = mean_for_sample[..., None] + tf.linalg.matmul(chol, self.eps)  # [..., N, L, S]
+            samples = leading_transpose(samples, [..., -1, -3, -2])  # [..., S, N, L]
+            samples = tf.linalg.adjoint(samples)  # [..., (S), N, L]
+
+        return samples
+
+    @abstractmethod
+    def prepare_acquisition_function(
+            self, dataset: Dataset, model: ModelInterface
+    ) -> AcquisitionFunction:
+        """
+        :param dataset: The data to use to build the acquisition function.
+        :param model: The model over the specified ``dataset``.
+        :return: An acquisition function.
+        """
+
+
+class MonteCarloExpectedImprovement(MonteCarloAcquisition):
+    """
+    Builder for the Monte_carlo based expected improvement.
+    """
+
+    def prepare_acquisition_function(
+            self, dataset: Dataset, model: ModelInterface
+    ) -> AcquisitionFunction:
+        """
+        :param dataset: The data from the observer.
+        :param model: The model over the specified ``dataset``.
+        :return: The expected improvement function.
+        """
+        mean, _ = model.predict(dataset.query_points)
+        eta = tf.reduce_min(mean, axis=0)
+        return lambda at: self._acquisition_function(self, model, eta, at)
+
+    @staticmethod
+    def _acquisition_function(
+            self, model: ModelInterface, eta: tf.Tensor, at: QueryPoints
+    ) -> tf.Tensor:
```

Comment:
i'm starting to think this `_acquisition_function` method style is just confusing. It's arbitrary how it's defined, and it doesn't return an `AcquisitionFunction` anyway

Reply:
isn't it used when we plot the acq fct?

Reply:
i don't think it's ever used